sway/tree/workspace.c

```diff
@@ -106,96 +106,104 @@ static bool workspace_valid_on_output(const char *output_name,
 	return true;
 }
 
+static void workspace_name_from_binding(const struct sway_binding * binding,
+		const char* output_name, int *min_order, char **earliest_name) {
+	char *cmdlist = strdup(binding->command);
+	char *dup = cmdlist;
+	char *name = NULL;
+
+	// workspace n
+	char *cmd = argsep(&cmdlist, " ");
+	if (cmdlist) {
+		name = argsep(&cmdlist, ",;");
+	}
+
+	if (strcmp("workspace", cmd) == 0 && name) {
+		char *_target = strdup(name);
+		_target = do_var_replacement(_target);
+		strip_quotes(_target);
+		while (isspace(*_target)) {
+			memmove(_target, _target+1, strlen(_target+1));
+		}
+		wlr_log(WLR_DEBUG, "Got valid workspace command for target: '%s'",
+				_target);
+
+		// Make sure that the command references an actual workspace
+		// not a command about workspaces
+		if (strcmp(_target, "next") == 0 ||
+			strcmp(_target, "prev") == 0 ||
+			strcmp(_target, "next_on_output") == 0 ||
+			strcmp(_target, "prev_on_output") == 0 ||
+			strcmp(_target, "number") == 0 ||
+			strcmp(_target, "back_and_forth") == 0 ||
+			strcmp(_target, "current") == 0)
+		{
```

Comment:
style: put this brace on the previous line

Reply:
Done; also, indented by two tabs as per style and since the structure is otherwise confusing.